packages/@ember/-internals/metal/lib/computed.ts

```diff
@@ -675,6 +676,25 @@ export class ComputedProperty extends ComputedDescriptor {
     let ret;
 
     try {
+      if (obj instanceof Ember.Component) {
+        // ensure two way binding works when the component has defined a computed
+        // property with both a setter and dependent keys, in that scenario without
+        // the sync observer added below the caller's value will never be updated
+        //
+        // See GH#18417 / GH#19028 for details.
+        let descriptor: ComputedDescriptor;
+        descriptor = descriptorForProperty(obj, keyName);
+        let meta = metaFor(obj);
+        if (
+          meta.isInitializing() &&
+          descriptor !== undefined &&
+          descriptor._dependentKeys !== undefined &&
+          descriptor._dependentKeys.length > 0
+        ) {
```

Comment:
I'm curious though, why are we doing `descriptorForProperty` at all. The class that this is in **is** the descriptor, see here for where that is setup:

https://github.com/emberjs/ember.js/blob/8b64cc52f828941667dc7ac267c46a2752bccf1f/packages/@ember/-internals/metal/lib/decorator.ts#L62-L69

So, I would expect that we can literally use `this` instead.